### PR TITLE
Updated user form to allow file uploads (and other files to handle file uploads)

### PR DESCRIPTION
**Pull request by jhuang09**
Changes made:
- MessageServlet.java:
    - included the blobstore library;
    - added some lines to handle converting the upload into a URL;
- Message.java:
    - added a `setImageUrl` function to set the newly added `image` attribute to a string specified by the user
- user-page-load.js
    - added a new function that will set the action of the form and will display the form if the user is logged in.
- user-page.html
    - made the messages form multipart!
    - included the option to specify a recipient to the message

While it is possible to upload and submit an image, the image itself isn't stored yet, so I will work on that soon! :)

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/ImageUploadUrlServlet.java
Comment (dhgarrette) [inline src/main/java/com/google/codeu/servlets/ImageUploadUrlServlet.java]: Extra line.

File: src/main/java/com/google/codeu/data/Message.java
Comment (dhgarrette) [inline src/main/java/com/google/codeu/data/Message.java]: Let's call this `imageUrl` so that it's clear what's being stored.

File: src/main/java/com/google/codeu/data/Message.java
Comment (dhgarrette) [inline src/main/java/com/google/codeu/data/Message.java]: `imageUrl` here too.

File: src/main/webapp/user-page.html
Comment (dhgarrette) [inline src/main/webapp/user-page.html]: This should be indented to line up with the `<form>` above.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java
Comment (dhgarrette) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java:147]: I think this new code would benefit from some comments.  Or maybe just pulling the new stuff into a separate method?

File: src/main/java/com/google/codeu/data/Message.java
Comment (dhgarrette) [inline src/main/java/com/google/codeu/data/Message.java]: Does this mean that a message can only have one image?

File: src/main/webapp/js/user-page-loader.js
Comment (dhgarrette) [inline src/main/webapp/js/user-page-loader.js]: This addition of a "recipient" field doesn't seem directly related to uploading images.  Should it be a separate PR?

**Review by dhgarrette — CHANGES_REQUESTED** [review]
Thanks, Janny!  I added a few comments, but when I try out the change with `mvn appengine:devserver`, the site crashes with a `java.lang.ClassNotFoundException: javax.activation.DataSource` error when I hit the Submit button, whether or not I'm uploading an image.  Are you getting the same?

File: src/main/java/com/google/codeu/data/Message.java
Comment (jhuang09) [reply src/main/java/com/google/codeu/data/Message.java]: I think that attribute is specifically for when the user is uploading a file, and I believe the user can only upload one file at a time.

File: src/main/webapp/js/user-page-loader.js
Comment (jhuang09) [reply src/main/webapp/js/user-page-loader.js]: That's super fair. I'm a bit stuck on how to create a separate PR for this. Should I only add and commit this file on another branch and then make a new PR?

**Comment by jhuang09** [comment]
Hey Dan! Sorry it took a while for me to get back to you! I don't have that error when I hit the "Submit" button. I'm not entirely sure why there's this difference. I'll make some commits, and see if anything has changed?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java
Comment (jhuang09) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java:147]: On it! Trying to figure out if passing a Message object as a parameter to another function and calling one of its set methods would update the Message object in the function it was called from.

Comment (jhuang09) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java:147]: I've added a getImageUrl function to help check this and it seems like it does properly update the Message object after the function terminates!

**Comment by jhuang09** [comment]
I did some Googling about the error, and it might be happening for computers that are running with Java 9. One of the solutions I found was to add a dependency to the `pom.xml` file. I can commit and push that change and see if that solves the issue?

File: src/main/java/com/google/codeu/data/Message.java
Comment (dhgarrette) [inline src/main/java/com/google/codeu/data/Message.java]: You've got a typo here.  It should be `imageUrl`.

File: src/main/webapp/user-page.html
Comment (dhgarrette) [inline src/main/webapp/user-page.html]: Looking at other fields in this file, it seems like the convention is for IDs to use all lowercase and hyphens between words.  So, following, e.g. line 44 which uses `message-input`, let's use `id="recipient-input"` for this.

File: src/main/webapp/js/user-page-loader.js
Comment (dhgarrette) [reply src/main/webapp/js/user-page-loader.js]: Yep, exactly, I think that would be the easiest way.  So you'd do something like:

```
git checkout master
git checkout -b recipient-field
# Make changes to files
git push --set-upstream origin recipient-field
```

And then create a PR for the new branch.  When that PR gets approved, you can merge it, and then switch back to this branch and pull in the changes:

```
git checkout JannyDevelopment
git pull origin master
```

File: src/main/java/com/google/codeu/servlets/MessageServlet.java
Comment (dhgarrette) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java:147]: Cool, yeah, Java objects are always references, so passing objects into functions won't copy them :-)

**Review by dhgarrette — CHANGES_REQUESTED** [review]
Oh, weird.  Problems between Java versions are always super annoying!  I'm apparently using Java 10, and I tried adding the following to the pom:

```
    <dependency>
      <groupId>javax.activation</groupId>
      <artifactId>activation</artifactId>
      <version>1.1.1</version>
    </dependency>
```

but it didn't solve my problem.

File: src/main/webapp/user-page.html
Comment (jhuang09) [reply src/main/webapp/user-page.html]: Thanks for letting me know! I'll update the names and push it to the PR that's concerned with the addition of a recipient field.

**Comment by dhgarrette** [comment]
Ok, I was able to install Java 8 and get this working.  Can you sync this branch (`git pull origin master`) and fix any remaining issues so we can verify that it works for Maria and Mo?

**Comment by jhuang09** [comment]
I've updated all the files! I'll get working on the section that stores the image. Should I wait until this branch has merged and make another PR for the storing portion of this feature?

**Comment by dhgarrette** [comment]
Cool!  There's one small thing to fix but I think it's in pretty good shape.  I'll ask Maria and Mo to see if they can get Java 8 set up, and if they do, then we can merge this.  Maybe in the meantime you can create a new branch and do the Event datastore stuff?

**Comment by jhuang09** [comment]
I realized that the `form` tag did not line up with its starting `form` tag, so I updated the file so they lined up. What was the small thing that needed to be fixed?

**Comment by dhgarrette** [comment]
Oh, yeah, that was the one thing.  I thought I made a comment on that line, but maybe it didn't save or something.  Oh well, you got it anyway :-)  Since Mo, Maria, and I were able to verify that this change is ok for them, I'll approve and you can merge :-)  Thanks for your patience!

**Review by dhgarrette — APPROVED** [review]

**Comment by jhuang09** [comment]
Woohoo! So exciting :D!

**Pull request merged**
